src/model/schema.js

```diff
@@ -193,11 +193,16 @@ export default class Schema {
 		this.decorate( 'checkAttribute' );
 
 		this.on( 'checkAttribute', ( evt, args ) => {
-			args[ 0 ] = new SchemaContext( args[ 0 ] );
+			if ( !( args[ 0 ] instanceof SchemaContext ) ) {
+				args[ 0 ] = new SchemaContext( args[ 0 ] );
```

Comment:
We could move it to the `SchemaContext` constructor. If the parameter is an instance of the schema context, the constructor should return this object.

Reply:
(in JS you can return from the class constructor)